### ISSUE TITLE
Stellar-MFN@ E31A not working properly / Display won't change.

### ISSUE THREAD

**Issue by ValiPR** [atc1441/ATC_TLSR_Paper#41]
Heya all!

I'm kinda new to all this so not sure where it went wrong or if my device just isn't compatible.

I flashed the ATC_Paper.bin to the Stellar-MFN@ E31A but the screen isn't doing anything and I am not able to chance anything, I am able to connect to it over bluetooth (ESL_FC09DB) But that's about it.
The LED flashes green occasionally. 

Excuse my shoddy way of attaching the pins hahaha.
![imgonline-com-ua-dexifvbItkFosYDyL](https://github.com/atc1441/ATC_TLSR_Paper/assets/141665563/602dc011-4c1b-4fb3-9e70-3ef49d30ea38)

If anyone has some ideas or could offer some help I would be really grateful!

**Comment by obones** [comment]
I'm using a MFN@ E31HA here and the result is about the same.
The display did get change a little bit, some white lines appeared in the middle of the existing price as displayed when it was in store, and there is a black rectangle flashing at the border of the display.

How should one go about sending an image to this display?

**Comment by ValiPR** [comment]
> I'm using a MFN@ E31HA here and the result is about the same. The display did get change a little bit, some white lines appeared in the middle of the existing price as displayed when it was in store, and there is a black rectangle flashing at the border of the display.
> 
> How should one go about sending an image to this display?

I actually managed to order a MN@ E31HA hoping it would be different and the result is I can flash it but the flash is "luck based", it shows the clock sometimes, but it's off center, using other users flashes nets the same result sadly :( If the flash fails, it shows maybe part of the clock, but the rest is all random "pixels" the Bluetooth capability also ceases to work.

**Comment by lvjiaming0418** [comment]
My m3n@ even doesn't have the bluetooth.
I cannot connect with it except the USB-COM Flash.

**Comment by TheoHarbers** [comment]
I have the same thing Stellar-M3N@ E31HA sn 301119. Flashing was without problem. I can talk to it over BLE but the display does not do very much. Sometimes after i sent some commands  and after 30 seconds or the display seems to produce a dark edge (couple of pixels wide) all around the outside stays visible and seems flashing a couple of seconds. I have a black-white-red version. I have tried the commands to change the display type manually, 0xE0020 or xE003 force set an EPD model, if it was not detect automatically correct. But no luck with that. any ideas are welcome.

**Comment by marekmeli** [comment]
@TheoHarbers I bought 20 of those exact units (M3N@ E31HA) and have troubles flashing them. During the UART flashing, the LED is blinking rapidly, but after finishing, the display remains unchanged and the BLE stays inactive - not possible to perform OTA.
How did you manage to flash it using UART successfully? Did you used [this](https://github.com/atc1441/ATC_TLSR_Paper/blob/main/USB_UART_Flashing_connection.jpg) connection diagram? Which flashing webpage did you use?
Thanks!

**Comment by TheoHarbers** [comment]
@marekmeli Yes I have used that connection diagram. It's almost 2 years ago and can't remember all the steps, but I just followed the information from internet/youtube. I did not spend anymore time on the project but it is still in my drawer. Perhaps there are updates to retry it some day.

**Comment by marekmeli** [comment]
@TheoHarbers Thank you, I will try a different UART adapter and will post here in case of any successful and reliable configuration.

**Comment by marekmeli** [comment]
@TheoHarbers I managed to do it all from scratch using this site: [https://atc1441.github.io/ATC_BLE_OEPL_Image_Upload.html](https://atc1441.github.io/ATC_BLE_OEPL_Image_Upload.html). I suggest moving towards OEPL since it has larger support and the possibility to connect with OpenEpaperLink.
With UART, select the Device Type to 45, then "Unlock Flash", "Erase All Flash", then select OEPL firmware and flash.

<img width="1497" height="1315" alt="Image" src="https://github.com/user-attachments/assets/e5ca275f-2da1-4cba-9d53-41ccedae35ef" />
After flashing, perform "Soft Reset MCU". After that, connect via BLE, and modify the HW setup according to this screenshot:

<img width="1515" height="1299" alt="Image" src="https://github.com/user-attachments/assets/bee5900d-4f95-45e0-aa2c-1bd3214b9ec1" />

I know that the panel is 12**2** by 250, but from my testing, setting the 120 brings more consistent results. Sometimes, the last 2 rows of pixels are messy, but fixing this would require building dedicated firmware.

Happy hacking!

**Comment by johnwhlee** [comment]
Hey @marekmeli,

Can I consult you on how you got it working for the Stellar-M3N@ E31HA? I've followed your settings, using the https://atc1441.github.io/ATC_BLE_OEPL_Image_Upload.html flasher. I see the LED blinking when it is flashing, but when it is completed it still shows me the same barcode with the ID of the device.

What UART adapter are you using? I'm using an FTDI FT232RL based adapter. You had a question about it earlier as well, was that the issue for you?

**Comment by marekmeli** [comment]
Hi @johnwhlee,

Yep, your referenced FT232RL is exactly the one that was not working. Even the symptoms were the same. Choose a different adapter, and you'll be alright :)

I'm using CP2102 STC based adapter (bought [this one](https://www.gme.cz/v/1508310/prevodnik-usb-uart-reset-pin) from the local eshop). This one is working flawlessly with all the default values (baud rate,...). Give it a go! Just be careful to buy a board with RST pin exposed. Fiddling with RST manually is a really painful process.

Let me know if you need further assistance.
